### PR TITLE
Correctly handle the bad usage of env in provider

### PR DESCRIPTION
**Pull request by Sytten**
Fixes https://github.com/prisma/vscode/issues/63

### CODE REVIEW

File: libs/datamodel/core/tests/config/sources.rs [hunk 1]
```diff
@@ -69,7 +69,6 @@ fn must_forbid_env_functions_in_provider_field() {
             url = env("DB_URL")
         }
     "#;
-    std::env::set_var("DB_PROVIDER", "postgresql");
```
Comment (mavilein) [inline libs/datamodel/core/tests/config/sources.rs:72]: I guess the test would fail if this line is present?

Comment (Sytten) [reply libs/datamodel/core/tests/config/sources.rs:72]: Its hard to make a test fail for that code really, but the test failed with the previous code without it.

Comment (Sytten) [reply libs/datamodel/core/tests/config/sources.rs:72]: Right, I will fix that.

Comment (Sytten) [reply libs/datamodel/core/tests/config/sources.rs]: Fixed @mavilein

**Pull request merged**